webapp/src/components/tutorial/TutorialValidationErrorMessage.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from "react";
+import { Button } from "../../../../react-common/components/controls/Button";
+import { Button as SuiButton } from "../../sui";
+import CodeValidationResult = pxt.tutorial.CodeValidationResult;
+
+interface TutorialValidationErrorMessageProps {
+  onContinueClicked: () => void;
+  onReturnClicked: () => void;
+  validationFailures: CodeValidationResult[];
+  tutorialId: string;
+  currentStep: number;
+}
+
+export function TutorialValidationErrorMessage(props: TutorialValidationErrorMessageProps) {
+  const [showHint, setShowHint] = React.useState(false);
+
+  const onShowHintClicked = () => {
+    if (!showHint) {
+      pxt.tickEvent("codevalidation.showhint", {
+        tutorial: props.tutorialId,
+        step: props.currentStep,
+      });
+      setShowHint(true);
+    }
+  };
+
+  const hintsExist = !!props.validationFailures.find(f => f.hint && f.hint != "");
+
+  const hintContent = hintsExist ? (
+    <div className="tutorial-validation-error-hint">
+      {props.validationFailures.map((f, i) => {
+        return f.hint ? <div key={i}>{f.hint}</div> : null;
+      })}
+    </div>
+  ) : undefined;
+
+  const showHintLabel = lf("Show Hint");
+  const continueAnywayLabel = lf("Continue Anyway");
+  const closeLabel = lf("Close");
+  return (
+    <div className="tutorial-validation-error-container">
+      {/* Using SuiButton to get close icon consistent with other popups */}
+      <SuiButton
```

Comment:
can we just match modal and use normal react-common button here instead of sui? https://github.com/microsoft/pxt/blob/master/react-common/components/controls/Modal.tsx#L100